contracts/dapis/DapiServer.sol

```diff
@@ -636,119 +636,117 @@ contract DapiServer is
 
     ///                     ~~~OEV~~~
 
-    /// @notice Updates a data feed that the OEV proxy reads using data signed
-    /// by the respective Airnodes for the specific bid. The Beacons for which
-    /// the fulfillment data and signature is omitted will be read from
-    /// storage.
-    /// @dev Even though data for Beacons are signed individually, the caller
-    /// is only allowed to use the signatures as a bundle. They cannot omit
-    /// individual signatures or mix-and-match among bundles.
+    /// @notice Updates a data feed that the OEV proxy reads using the
+    /// aggregation signed by the absolute majority of the respective Airnodes
+    /// for the specific bid
+    /// @dev For when the data feed being updated is a Beacon set, an absolute
+    /// majority of Airnodes must sign the resulting value and timestamp. While
+    /// doing so, Airnodes should refer to data signed by the respective
+    /// Airnodes. The absolute majority of this signed data should have similar
+    /// timestamps and values (e.g., within 2 minutes and 1% deviation).
+    /// Freshness should not be required, as OEV proxy data feeds cannot be
+    /// updated using older timestamps.
     /// @param oevProxy OEV proxy that reads the data feed
+    /// @param dataFeedId Data feed ID
     /// @param updateId Update ID
-    /// @param signatureCount Number of signatures in `signedData`
-    /// @param signedData Array of ABI-encoded Airnode address, template ID,
-    /// timestamp, fulfillment data and bid metadata that is signed by the
-    /// respective Airnode for the specific bid
+    /// @param timestamp Signature timestamp
+    /// @param data Update data (an `int256` encoded in contract ABI)
+    /// @param packedOevUpdateSignatures Packed OEV update signatures, which
+    /// include the Airnode address, template ID and these signed with the OEV
+    /// update hash
     function updateOevProxyDataFeedWithSignedData(
         address oevProxy,
+        bytes32 dataFeedId,
         bytes32 updateId,
-        uint256 signatureCount,
-        bytes[] calldata signedData
-    ) external payable override {
-        uint256 beaconCount = signedData.length;
-        bytes32 metadataHash = keccak256(
+        uint256 timestamp,
+        bytes calldata data,
+        bytes[] calldata packedOevUpdateSignatures
+    ) external payable override onlyValidTimestamp(timestamp) {
```

Comment:
`onlyValidTimestamp()` is used because we don't want an update that has incorrectly used a wrong timestamp to permanently break the data feed. That is a bit risky because doing so will also block correct updates if there is chain time drift. `onlyValidTimestamp()` requires the timestamp to not be from the future more than 1 hour, assuming the chain time drift will be less than that.

The contract checking signature freshness is susceptible to blocking honest updates when there is chain time drift. When I said "we don't need to worry about chain time drift" I meant it in the sense that there won't be a freshness check. However, this doesn't make sense, because the problem with the contract checking signature freshness is that it checks this against the block time (which may drift), while the Airnode will be checking against the system time, so this is not a concern. I don't see why we would want to auction off outdated updates, so I will modify the comments to say that Airnodes will require the aggregation to be fresh to sign the respective OEV update.